### PR TITLE
Per key/layer RGB underglow

### PR DESCRIPTION
**Pull request by darknao**
This adds per key / per layer RGB underglow configuration.
It's mainly based on @valdur mod with additional changes to make it configurable via devicetree.

It includes:
- [Valdur's RGB mod](https://github.com/moergo-sc/zmk/compare/main...valdur:zmk-glove80:valdur-stuff)
- Refactor, move configuration to devicetree, and some battery life optimization
- fix(rgb): auto-off logic [#2244](https://github.com/zmkfirmware/zmk/pull/2244) from ZMK upstream
- Correctly set underglow state when entering sleep [#2284](https://github.com/zmkfirmware/zmk/pull/2284) from ZMK upstream
- feat(split): Increase split interval during idle [#682](https://github.com/zmkfirmware/zmk/pull/682) from ZMK upstream

How to use:
More instructions are available on my [config repo](https://github.com/darknao/glove80-zmk-config).
TLDR: Add and customize this snippet to the Custom Device-tree field in the layout editor:

```
#include <dt-bindings/zmk/rgb_colors.h>
/ {
    underglow-layer {
        compatible  = "zmk,underglow-layer";

        // example of RGB layer for the Lower keymap layer
        lower {
            bindings = <
                ______ ______ ______ ______ ______                                                          ______ ______ ______ ______ ______
                PURPLE   PINK   PINK   PINK   PINK   PINK                                            ______   GOLD ORANGE ORANGE ORANGE    RED
                PURPLE ______ ______ ORANGE ______ ______                                            ______ YELLOW YELLOW YELLOW ORANGE ______
                PURPLE ______ ORANGE    RED ORANGE ______                                            ______ YELLOW YELLOW YELLOW ORANGE ______
                  BLUE ______    RED    RED    RED ______ ______ ______ ______  ______ ______ ______ ______ YELLOW YELLOW YELLOW    RED   BLUE
                ______ ______ ORANGE ORANGE ORANGE        ______  GREEN ______  ______ ______ ______        YELLOW YELLOW YELLOW    RED ______
                >;
            layer-id = <LAYER_Lower>;
        };

        // This defines another RGB layer
        another-layer {
            bindings = < [...] >;
            layer-id = <LAYER_Name>;
        };
    };
};
```
`______` means the RGB is off. You can use [predefined color name](https://github.com/darknao/zmk/blob/darknao/rgb-dts/app/include/dt-bindings/zmk/rgb_colors.h) or RGB hex code like `0xFF0000` for red.
Full example on the Layout Editor [here](https://my.glove80.com/#/layout/user/98218d07-c187-4a97-b020-f5df021abe73)

To preserve battery life, I recommend enabling the `ZMK_RGB_UNDERGLOW_AUTO_OFF_IDLE` KConfig to automatically turn off underglow after `ZMK_IDLE_TIMEOUT` ms (default to 30 seconds).
From my limited experience, you can expect around one week of battery life with no RGB on the base layer.

### CODE REVIEW

**Comment by darknao** [comment]
Closing as it has been replaced with #36
